src/video_utils.py

```
@@ -164,13 +166,74 @@ def get_video_duration(video_path):
+        url,
+        "-d",
+        temp_dir,
+    ]
+
+    try:
+        debug("Running gallery-dl command: %s", command)
+        subprocess.run(command, check=True, timeout=120)
+        result_path = []  # Initialize the result variable as a empty list
+        # Use Path.rglob to recursively search for files in the temp directory
+        # as the output contains subdirectories and may contain multiple files
+        for file in [str(file) for file in Path(temp_output).rglob("*")]:
```

Ruff (0.8.2)

Undefined name `temp_output`

(F821)